### PR TITLE
correctly handle path escaping for connector IDs

### PR DESCRIPTION
**Pull request by bobcallaway**
Fixes: #2289 

Signed-off-by: Bob Callaway <bob.callaway@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note
Correctly handle path escaping for connector IDs
```

### CODE REVIEW

**Review by nabokihms — COMMENTED** [review]
Good catch! Thank you a lot. I have only one moment to discuss, and then we will be able to move further with it.

File: server/server.go [hunk 1]
```diff
@@ -302,7 +302,7 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 		}
 	}
 
-	r := mux.NewRouter()
+	r := mux.NewRouter().SkipClean(true).UseEncodedPath()
```
[the remark this answers is not in the document]
Comment (bobcallaway) [reply server/server.go:305]: Would you want to always encode the connector ID or only if it has non-URL safe characters in it? I imagine there is some UX benefit to leaving it as an unencoded string if the connector ID value is already URL-safe.

[the remark this answers is not in the document]
Comment (nabokihms) [reply server/server.go:305]: We do not want users to directly request URLs with the connector name in them (and I don't think it is possible right now). The recommended way is to request the `/auth` endpoint with the `connector_id` query parameter. Thus it is not a big deal to keep per connector auth URLs with ids encoded.

**Comment by sagikazarmark** [comment]
As far as I can tell there are changes requested, so moving to next release.

**Comment by sagikazarmark** [comment]
@bobcallaway @nabokihms any update about the status of this PR? From what I can tell there is a debate about the implementation.

**Comment by nabokihms** [comment]
We can still accept the current implementation. The debate is about details o internal component implementation and only about cosmetics.

@bobcallaway, I think we can accept this PR as is, yet rebasing the branch is still required.

**Comment by bobcallaway** [comment]
ok, i will rebase it then

**Review by nabokihms — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**